docs/development/testguide.rst

```diff
@@ -764,66 +764,47 @@ Running image tests
 We make use of the `pytest-mpl <https://pypi.org/project/pytest-mpl>`_
 plugin to write tests where we can compare the output of plotting commands
 with reference files on a pixel-by-pixel basis (this is used for instance in
-:ref:`astropy.visualization.wcsaxes <wcsaxes>`).
+:ref:`astropy.visualization.wcsaxes <wcsaxes>`). We use the `hybrid mode
+<https://github.com/matplotlib/pytest-mpl#hybrid-mode-hashes-and-images>`_ with
+hashes and images.
 
-To run the Astropy tests with the image comparison, use::
+To run the Astropy tests with the image comparison, use e.g.::
 
-    pytest --mpl --remote-data=astropy
+    tox -e py39-test-image-mpl311
 
-However, note that the output can be very sensitive to the version of Matplotlib
-as well as all its dependencies (e.g., freetype), so we recommend running the
-image tests inside a `Docker <https://www.docker.com/>`__ container which has a
-frozen set of package versions (Docker containers can be thought of as mini
-virtual machines). See our ``.circleci/config.yml`` for reference.
+However, note that the output can be very sensitive to the specific version of
+libraries such as freetype, as well as the operating system, so we do not
+recommend running the image tests locally in general and instead it is best to
+rely on these running in an controlled continuous integration environment.
 
 Writing image tests
 -------------------
 
 The `README.rst <https://github.com/matplotlib/pytest-mpl/blob/master/README.rst>`__
-for the plugin contains information on writing tests with this plugin. The only
-key addition compared to those instructions is that you should set
-``baseline_dir``::
+for the plugin contains information on writing tests with this plugin. Once you
+have added a test, and push this to a pull request, you will likely start seeing
+a test failure because the figure hash is missing from the hash libraries
+(see the next section for how to proceed).
 
-    from astropy.tests.image_tests import IMAGE_REFERENCE_DIR
+Failing tests
+-------------
 
-    @pytest.mark.mpl_image_compare(baseline_dir=IMAGE_REFERENCE_DIR)
+When existing tests start failing, it is usually either because of a change in
+astropy itself, or a change in Matplotlib. New tests will also fail if you have
+not yet updated the hash library.
 
-This is because since the reference image files would contribute significantly
-to the repository size, we instead store them on the http://data.astropy.org
-site. The downside is that it is a little more complicated to create or
-re-generate reference files, but we describe the process here.
+In all cases, you can view a webpage with all the existing figures where you can check whether
```

Comment:
Should we add some indication of where to find this magical page here?

Reply:
Done